### ISSUE TITLE
Converting NSDate to Instant and back to NSDate resulting not equal NSDate

### ISSUE THREAD

**Issue by MAX-POLKOVNIK** [Kotlin/kotlinx-datetime#188]
Hi everyone. Thanks for great library. 
I have a next code: 
```
fun testEquals(date: NSDate): Boolean {
        val instant = date.toKotlinInstant()
        val date2 = instant.toNSDate()
        return date == date2
}
```
Which returns false. 
Expected to be true.

Library version - 0.3.2

**Comment by dkhalanskyjb** [comment]
The KDoc for `NSDate.toKotlinInstant` states:
> Even though Darwin only uses millisecond precision, it is possible that [date] uses larger resolution, storing microseconds or even nanoseconds. In this case, the sub-millisecond parts of [date] are rounded to the nearest millisecond, given that they are likely to be conversion artifacts.

Does this explain the behavior you're seeing? Also, I didn't check if that's the case, but it's possible that `NSDate` values are compared as floating-point values, which would also break due to even slight manipulations. https://developer.apple.com/documentation/foundation/nsdate/1416400-isequal seems to imply that this is the case, but, again, not sure.

**Comment by MAX-POLKOVNIK** [comment]
Yes, that's issue. Just checked via timeIntervalSince1970. Source date is 1646895805.931058 and after conversion it is 1646895805.931. Will it be fixed or is there workaround to prevent that conversion artifact?

**Comment by dkhalanskyjb** [comment]
The Foundation library itself only treats `NSDate` as having millisecond precision. Perform any operation with an `NSDate`, and the sub-millisecond part gets discarded. So, anything more granular than milliseconds doesn't have a meaning. If you need better precision, `NSDate` is not suitable.
